### PR TITLE
Android support

### PR DESCRIPTION
**Pull request by vmkononenko**
This changes make uperf being able to run on Android platform. Enhancements for sleep operations are implemented to reduce uperf impact on device power consumption during tests.

### CODE REVIEW

File: src/delay.c [hunk 1]
```diff
@@ -20,43 +20,101 @@
...
 
-#include <time.h>
-#include <math.h>
-#ifdef HAVE_CONFIG_H
-#include "../config.h"
-#endif /* HAVE_CONFIG_H */
+#include "uperf.h"
+#include "logging.h"
+
+#include <hardware_legacy/power.h>
```
Comment (uperf) [inline src/delay.c]: Should this be behind a #ifdef UPERF_ANDRIOD ?

File: src/delay.h [hunk 2]
```diff
@@ -28,4 +28,9 @@
 int uperf_sleep(hrtime_t);
 int uperf_spin(hrtime_t);
 
+/*
+ * Initialize alarm to be able to wake up the device between transactions / flowops
+ */
+int init_android_alarm();
```
Comment (uperf) [inline src/delay.h]: Should this also be behind a #ifdef UPERF_ANDROID?

File: src/flowops_library.c [hunk 3]
```diff
@@ -22,7 +22,7 @@
  */
 
 #ifdef HAVE_CONFIG_H
-#include "config.h"
+#include "../config.h"
```
Comment (uperf) [inline src/flowops_library.c]: I do not think this is required.. We use -I. -I.. while building so the header is picked up. Can you pass the same -I flags to the android compiler?

File: src/delay.c (hunk 1 above)
Comment (vmkononenko) [inline src/delay.c]: Right, will fix.

File: src/delay.h (hunk 2 above)
Comment (vmkononenko) [inline src/delay.h]: For now yes, will put it under ifdef. Maybe in future we'll want to use alarms for sleeping on Linux as well. Then device under test will be able to suspend during think operations.

File: src/flowops_library.c (hunk 3 above)
Comment (vmkononenko) [inline src/flowops_library.c]: Before I made this change I looked through uperf code and checked how config is included. In most places it is included via path specification so I changed here for consistency.

Comment (vmkononenko) [inline src/flowops_library.c]: Fixed

**Pull request merged**